lessons/session_5/session_5_model_development.md

```diff
@@ -0,0 +1,37 @@
+---
+marp: true
+theme: uncover
+_class: invert
+paginate: true
+output: pdf_document
+---
+
+# ML Systems in Production
+# Session 5: Model Development and Evaluation
+Jesús A. Calderón
+February 2024
+
+---
+
+# Agenda
+
+**5.1 Model Development and Offline Evaluation**
+	
++ Model Development and Training 
++ Ensembles 
++ Experiment Tracking and Versioning 
++ Distributed Training
++ AutoML 
++ Model Offline Evaluation 
+	
+
+---
+
+# Agenda (cont.)
+
+**Assignment: Setting Up a Code Repository**
+
++ Implement DB with Docker
```

Comment:
As you are mentioning the usage of Docker, is the installation of Docker a pre req of this course? Should we have learners install Docker when we host the tech onboarding sessions?

Reply:
Ideally, yes. I mentioned to Thomas that I would try to work with a SQLite backend, but I think there is a benefit in using a more solid backend like Postgres or MySql.  Also, offering the experience of working with Docker is beneficial and would be a good highlight to have in a resume.

Reply:
I agree with the benefits, but we also need to factor in the complexity and timeline we are faced with. This needs to be discussed in a meeting.

Reply:
Absolutely. Just for sake of  the thread:

+ The idea is to use [Sacred](https://sacred.readthedocs.io/en/stable/index.html) for experiment tracking.
+ Sacred supports various backends, including file-based, SQL, and MongoDB. In principle, we could do without a SQL server and rely on SQLite or text files to go through experiment tracking.
+ If we had Docker, though, we could also implement a MongoDB and discuss the benefits of one tool vs the other. 

Let me know when it would be a good time to talk. As I said, I do not think that removing Docker is a showstopper, but it is a good addition to the curriculum.

I'll make a commit with this remove to get the change through.

Reply:
Done. Removed.

Reply:
@calderonjesus Feel free to put it back or if you don't think it is necessary, let me know to merge the PR in.

Reply:
Brilliant. Thank you for the note, Daniel. I will put Docker back in the next commit.